extensions/python.bzl

```diff
@@ -0,0 +1,282 @@
+"""Python requirements module extension."""
+
+load(
+    "//:defs.bzl",
+    package_annotation = "py_package_annotation",
+)
+load(
+    "//private:reqs_repo.bzl",
+    "BUILD_FILE_TEMPLATE",
+    "RULES_PYTHON_COMPAT",
+    "generate_interface_bzl_content",
+    "process_lockfile",
+    "write_defs_file",
+)
+load("//private:sdist_repo.bzl", "sdist_repository")
+load("//private:whl_repo.bzl", "whl_repository")
+
+def _is_wheel(data):
+    if "url" in data and data["url"]:
+        if ".whl" in data["url"]:
+            return True
+        return False
+
+    if "whl" in data and data["whl"]:
+        return True
+
+    return False
+
+def _reqs_hub_impl(repository_ctx):
+    repository_ctx.file(
+        "BUILD.bazel",
+        BUILD_FILE_TEMPLATE.format(
+            packages = json.encode_indent(
+                sorted(depset(repository_ctx.attr.packages).to_list()),
+            ),
+        ),
+    )
+    for defs_id in repository_ctx.attr.defs:
+        defs_file = repository_ctx.path("defs_{}.bzl".format(defs_id))
+        write_defs_file(
+            repository_ctx,
+            repository_ctx.attr.packages,
+            defs_file,
+            defs_id,
+            name = repository_ctx.attr.hub_name,
+        )
+
+    repository_ctx.file(
+        "requirements.bzl",
+        RULES_PYTHON_COMPAT.format(
+            repository_name = repository_ctx.name,
+        ),
+    )
+    repository_ctx.file(
+        "defs.bzl",
+        generate_interface_bzl_content(repository_ctx.attr.defs, repository_ctx.name),
+    )
+
+reqs_hub = repository_rule(
+    implementation = _reqs_hub_impl,
+    attrs = {
+        "hub_name": attr.string(),
```

Comment:
You should fill in `doc` and `mandatory`

Reply:
Deleted this repo